var/spack/repos/builtin/packages/pumi/package.py

```diff
@@ -125,7 +126,14 @@ def test(self):
         description = "testing pumi uniform mesh refinement"
         self.run_test(exe, options, expected, purpose=description)
 
-        mpiexec = self.spec["mpi"].prefix.bin.mpiexec
+        mpiexec = ""
+        mpiexe_list = ["mpirun", "mpiexec", "srun"]
+        for mpiexe in mpiexe_list:
+            if which(mpiexe) is not None:
+                mpiexec = Executable(mpiexe).command
+                break
+        if mpiexec == "":
+            warnings.warn("MPI exec not found")
```

Comment:
I am unsure if this is how I should print the warning message. When I run the test with this warning nothing prints. I am still looking into it. Do you know the answer here?

Reply:
@Angelyr Good question. I'm not sure.  I see a lot of packages calling `print(...)` :

```
~/develop/spack/var/spack/repos/builtin/packages (develop)$ git grep -i print | grep "package\.py"
```

Tangentially related, others are using `raise RunTimeError(...)`: 

```
~/develop/spack/var/spack/repos/builtin/packages (develop)$ git grep -i Runtime | grep "package\.py"
```

I didn't see anything obvious in the spack docs (https://spack.readthedocs.io/en/latest/index.html).

Reply:
print didn't work for me either. We could merge and see what the spack people think